### DISCUSSION TITLE
Writes/changes not automatically detected

### DISCUSSION THREAD

**Discussion by jfathi** [mkdocs/mkdocs#2437] [Q&A]
Hello! I'm just starting to use mkdocs and have an odd error.

When the application is live on port 8000 (via `mkdocs serve`), it should watch for changes in the directory and dynamically update as necessary. However, when I make changes to the files (e.g.: index.md, mkdocs.yml), the site remains static. Here, the handler for detecting changes - in my case, handlers:64 - does not start.

I pulled the open-source Docker image (squidfunk/mkdocs-material) to correct the issue like so:
`docker run --rm -it -p 8000:8000 -v "$(pwd):/docs" squidfunk/mkdocs-material`

However, I'd like to run the app from my root machine.

Any thoughts on how to fix this? I am working on a Win10+WSL system, using Vim/VS Code to modify/write to directory files. Thanks!

**Comment by waylan** [comment]
> the handler for detecting changes - in my case, handlers:64 - does not start.

Are receiving any errors? Without more information, there's not any help we can provide. It should "just work."

That said, we are completely replacing the server in the next release (which should happen within the next week or so), so any issues with the current server will not likely carry over to the new one.

**Reply by jfathi** [reply]
That's what's odd - no errors are showing. I'll wait for the next release and report if the errors persist.
Thanks!

**Reply by jfathi** [reply]
Issue persists after installing new release (1.2.1) to a virtual env. Documentation is generated, but doesn't update.
Commands `mkdocs serve` and `python -m mkdocs serve` both have error.
![image](https://user-images.githubusercontent.com/57191828/121605578-a61fe280-ca1a-11eb-9e7f-c78ceb315020.png)

List of dependencies installed via pip:
![image](https://user-images.githubusercontent.com/57191828/121605618-ba63df80-ca1a-11eb-80d9-5bd06c07f9a5.png)

**Reply by jfathi** [reply]
Edit: **Issue is resolved.** All it took was a uninstall and reinstall of Python 3.9. 
Sometimes, the best solution is the easiest one.
